pkg/csi/service/vanilla/controller.go

```diff
@@ -446,13 +501,50 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		return nil, err
 	}
+	var volumePath string
+
+	// in-tree volume support
+	var isVolumePath bool
+	var volumeID string
+	if strings.Contains(req.VolumeId, ".vmdk") {
+		volumePath = req.VolumeId
+		isVolumePath = true
+	}
+	if c.manager.CnsConfig.FeatureStates.CSIMigration {
+		// Migration feature switch is enabled
+		if isVolumePath {
+			req.VolumeId, err = volumeMigrationService.GetVolumeID(ctx, volumePath)
+			if err != nil {
+				msg := fmt.Sprintf("failed to get VolumeID from volumeMigrationService for volumePath: %q", volumePath)
+				log.Error(msg)
+				return nil, status.Errorf(codes.Internal, msg)
+			}
+		}
+	} else {
+		// Migration feature switch is disabled
+		if isVolumePath {
```

Comment:
How will DeleteVolume be called for an in-tree volume if the feature is not enabled? Doesn't the user need to go through the whole upgrade path etc before getting here

Reply:
Wouldn't the delete anyways fail if migration FSS is disabled in the driver?